Python/symtable.c

```diff
@@ -1174,8 +1174,9 @@ symtable_visit_stmt(struct symtable *st, stmt_ty s)
             if (cur < 0) {
                 VISIT_QUIT(st, 0);
             }
-            if ((cur & (DEF_GLOBAL | DEF_NONLOCAL))
-                && s->v.AnnAssign.simple) {
+            if (((cur & DEF_NONLOCAL) |
+                 ((cur & DEF_GLOBAL) && !(cur & DEF_GLOBAL_TOP))
+                ) && s->v.AnnAssign.simple) {
```

Comment:
Adding a new symbol table flag to solve this looks like an overkill (plus TBH I don't fully understand this solution). Why not just skip emitting this error at global scope? For example check that `st->st_cur->ste_symbols != st->st_global`?

Reply:
Thanks for the review @ilevkivskyi. This is the first time I touch the symtable so sorry if the original proposed solution was a bit convoluted. I have simplified it in 701bffe.